app/src/main/java/org/wikipedia/dataclient/okhttp/CommonHeaderRequestInterceptor.kt

```diff
@@ -4,20 +4,25 @@ import okhttp3.Interceptor
 import okhttp3.Response
 import org.wikipedia.WikipediaApp
 import org.wikipedia.dataclient.RestService
+import org.wikipedia.login.LoginClient
 import java.io.IOException
 
 internal class CommonHeaderRequestInterceptor : Interceptor {
     @Throws(IOException::class)
     override fun intercept(chain: Interceptor.Chain): Response {
         val app = WikipediaApp.instance
+        val url = chain.request().url.toString()
         val builder = chain.request().newBuilder()
                 .header("User-Agent", app.userAgent)
                 .header("X-WMF-UUID", app.appInstallID)
-        if (chain.request().url.encodedPath.contains(RestService.PAGE_HTML_ENDPOINT)) {
+        if (url.contains(RestService.PAGE_HTML_ENDPOINT)) {
             builder.header("Accept", RestService.ACCEPT_HEADER_MOBILE_HTML)
-        } else if (chain.request().url.host.contains("maps.wikimedia.org")) {
+        } else if (url.contains("maps.wikimedia.org")) {
             builder.header("Referer", "https://maps.wikimedia.org/")
         }
+        if (LoginClient.enqueueForceEmailAuth && url.contains("action=clientlogin")) {
+            builder.removeHeader("User-Agent")
```

Comment:
Again, unclean logic for "forcing" email authentication. It looks like, if the server detects the apps' user-agent (`WikipediaApp/...`) then it's more likely to _ignore_ email authentication. To force it to trigger, let's just remove the user-agent entirely.